core/src/main/java/io/opentelemetry/android/export/RateLimitedExporter.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.opentelemetry.android;
+
+import android.util.Log;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+class BandwidthThrottlingExporter implements SpanExporter {
+    private final SpanExporter delegate;
+    private final Function<SpanData, String> categoryFunction;
+    private final long maxBytesPerSecond;
+    private final long timeWindowInMillis;
+    private long lastExportTime;
+    private long bytesExportedInWindow;
+
+    private BandwidthThrottlingExporter(Builder builder) {
+        this.delegate = builder.delegate;
+        this.categoryFunction = builder.categoryFunction;
+        this.maxBytesPerSecond = builder.maxBytesPerSecond;
+        this.timeWindowInMillis = builder.timeWindow.toMillis();
+        this.lastExportTime = System.currentTimeMillis();
```

Comment:
Use https://github.com/open-telemetry/opentelemetry-android/blob/e785fc627e688fa664f21ba5e68bebae98d07ca2/core/src/main/java/io/opentelemetry/android/internal/tools/time/SystemTime.kt#L31 for `System.currentTimeMillis()` so you can unit test the changes

Reply:
Also it seems like i cannot use return in an assignment statement, it doesn't pass spotless test. Or should i add and ignore the build test error?

Reply:
You should be able to just do `SystemTime.get().getCurrentTimeMillis()`, what s the error?